### DISCUSSION TITLE
Oauth Access Token Request Returning HTML Response

### DISCUSSION THREAD

**Discussion by ng-druid** [community/community#31485] [Apps, API and Webhooks]
We are currently trying to create a github app. However, have ran into a road block. When calling the endpoint https://github.com/login/oauth/access_token with the below credentials a HTML response is being returned rather than an expected JSON response. The end-point being called below is just a proxy to circumvent CORs. Also, the client_secret will eventually be moved to the server for security purposes. Just trying to get this working for a single account at this alpha stage. We did try removing the redirect url since it doesn't make sense to use to have a redirect url for a post request expecting to return a JSON response. That didn't appear to have any affect.

```typescript
    const urlParams = new URLSearchParams(window.location.search);
    const code = urlParams.get('code');
    const state = urlParams.get('state');
    const body = {
      client_id: ' Iv1.c72ed0518c7be356',
      client_secret: 'XXXXXXX',
      code,
      //redirect_uri: 'http://localhost:4000/github-callback2',
      state
    };
    const siteIndex = this.defaultDataServiceConfig.root.indexOf('/vertigoapp');
    const api = this.defaultDataServiceConfig.root.substring(0, siteIndex);
    console.log('api post', api);
    this.http.post(`${api}/github/login/oauth/access_token`, body).pipe(
      catchError(e => {
        console.log('github error', e);
        return of({});
      }),
      tap(res => {
        console.log('github response', res);
      })
    ).subscribe();
```

```go
func GithubAccessTokenRequest(req *events.APIGatewayProxyRequest) (string, error) {
	res, err := http.Post("https://github.com/login/oauth/access_token", "application/json", strings.NewReader(req.Body))
	if err != nil {
		return "", err
	}
	body, _ := ioutil.ReadAll(res.Body)
	log.Print(string(body))
	return string(body), nil
}
```

Response example: 
<img width="1440" alt="Screen Shot 2022-08-31 at 8 36 20 AM" src="https://user-images.githubusercontent.com/73197190/187684729-846e3257-533f-43c8-b6bb-704f7e9bf0a4.png">

**Comment by byrneh** [comment]
Hi @ng-druid [https://github.com/login/oauth/access_token ](https://github.com/login/oauth/access_token ) is not an API endpoint.  It is web GUI (browser) login page which is session and browser cookie based. 
API endpiybtsa are defined in docs here
[https://docs.github.com/en/rest](https://docs.github.com/en/rest)
[https://docs.github.com/en/graphql](https://docs.github.com/en/graphql)
